pinot-query-runtime/src/test/java/org/apache/pinot/query/runtime/QueryRunnerTest.java

```diff
@@ -200,6 +200,16 @@ private Object[][] provideTestSql() {
         new Object[]{"SELECT a.col1, b.col2 FROM a JOIN b ON a.col1 = b.col1 "
             + " WHERE a.col1 IN ('foo') AND b.col2 NOT IN ('')"},
 
+        // Range conditions with continuous and non-continuous range.
+        new Object[]{"SELECT a.col1, b.col2 FROM a JOIN b ON a.col1 = b.col1 "
+            + " WHERE a.col3 IN (1, 2, 3) OR (a.col3 > 10 AND a.col3 < 50)"},
+
+        new Object[]{"SELECT col1, SUM(col3) FROM a WHERE a.col3 BETWEEN 23 AND 36 "
+            + " GROUP BY col1 HAVING SUM(col3) > 10.0 AND MIN(col3) <> 123 AND MAX(col3) BETWEEN 10 AND 20"},
```

Comment:
In the current engine non-literals (function expressions) on RHS get rewritten by moving to LHS. Do we support that here ?

`WHERE a > b + 20` -> `WHERE a - b > 20`

Reply:
yes. this is not done by the range predicate but by the compilation on server.